include/sundials/sundials_adjointstepper.h

```diff
@@ -0,0 +1,81 @@
+/* -----------------------------------------------------------------
+ * SUNDIALS Copyright Start
+ * Copyright (c) 2002-2025, Lawrence Livermore National Security
+ * and Southern Methodist University.
+ * All rights reserved.
+ *
+ * See the top-level LICENSE and NOTICE files for details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ * SUNDIALS Copyright End
+ * -----------------------------------------------------------------
+ * SUNAdjointStepper class definition.
+ * ----------------------------------------------------------------*/
+
+#ifndef _SUNADJOINT_STEPPER_H
+#define _SUNADJOINT_STEPPER_H
+
+#include <sundials/sundials_adjointcheckpointscheme.h>
+#include <sundials/sundials_matrix.h>
+#include <sundials/sundials_nvector.h>
+#include <sundials/sundials_stepper.h>
+#include "sundials/sundials_types.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+typedef int (*SUNAdjRhsFn)(sunrealtype t, N_Vector y, N_Vector sens,
+                           N_Vector sens_dot, void* user_data);
+
+typedef struct SUNAdjointStepper_* SUNAdjointStepper;
+
+SUNDIALS_EXPORT
+SUNErrCode SUNAdjointStepper_Create(
+  SUNStepper fwd_sunstepper, sunbooleantype own_fwd, SUNStepper adj_sunstepper,
+  sunbooleantype own_adj, suncountertype final_step_idx, sunrealtype tf,
+  N_Vector sf, SUNAdjointCheckpointScheme checkpoint_scheme, SUNContext sunctx,
+  SUNAdjointStepper* adj_stepper);
+
+SUNDIALS_EXPORT
+SUNErrCode SUNAdjointStepper_ReInit(SUNAdjointStepper adj, N_Vector y0,
+                                    sunrealtype t0, N_Vector sf, sunrealtype tf);
```

Comment:
Match the style in the other integrators and `SUNStepper` (`t` first)
```suggestion
SUNDIALS_EXPORT
SUNErrCode SUNAdjointStepper_ReInit(SUNAdjointStepper adj, sunrealtype t0
                                    N_Vector y0, sunrealtype tf, N_Vector sf);
```